ush/parsing_namelists_CICE.sh

```diff
@@ -37,374 +38,72 @@ if (( $(( NY_GLB % NPY )) == 0 )); then
-   f_daidtd        = 'mdhxx'
-   f_dagedtt       = 'x'
-   f_dagedtd       = 'x'
-   f_mlt_onset     = 'mdhxx'
-   f_frz_onset     = 'mdhxx'
-   f_hisnap        = 'x'
-   f_aisnap        = 'x'
-   f_trsig         = 'x'
-   f_icepresent    = 'x'
-   f_iage          = 'x'
-   f_FY            = 'x'
-   f_aicen         = 'x'
-   f_vicen         = 'x'
-   f_vsnon         = 'x'
-   f_snowfracn     = 'x'
-   f_keffn_top     = 'x'
-   f_Tinz          = 'x'
-   f_Sinz          = 'x'
-   f_Tsnz          = 'x'
-   f_fsurfn_ai     = 'x'
-   f_fcondtopn_ai  = 'x'
-   f_fmelttn_ai    = 'x'
-   f_flatn_ai      = 'x'
-   f_fsensn_ai     = 'x'
-/
-
-&icefields_mechred_nml
-   f_alvl         = 'x'
-   f_vlvl         = 'x'
-   f_ardg         = 'x'
-   f_vrdg         = 'x'
-   f_dardg1dt     = 'x'
-   f_dardg2dt     = 'x'
-   f_dvirdgdt     = 'x'
-   f_opening      = 'x'
-   f_ardgn        = 'x'
-   f_vrdgn        = 'x'
-   f_dardg1ndt    = 'x'
-   f_dardg2ndt    = 'x'
-   f_dvirdgndt    = 'x'
-   f_krdgn        = 'x'
-   f_aparticn     = 'x'
-   f_aredistn     = 'x'
-   f_vredistn     = 'x'
-   f_araftn       = 'x'
-   f_vraftn       = 'x'
-/
-
-&icefields_pond_nml
-   f_apondn       = 'x'
-   f_apeffn       = 'x'
-   f_hpondn       = 'x'
-   f_apond        = 'mdhxx'
-   f_hpond        = 'mdhxx'
-   f_ipond        = 'mdhxx'
-   f_apeff        = 'mdhxx'
-   f_apond_ai     = 'x'
-   f_hpond_ai     = 'x'
-   f_ipond_ai     = 'x'
-   f_apeff_ai     = 'x'
-/
-
-&icefields_drag_nml
-   f_drag         = 'x'
-   f_Cdn_atm      = 'x'
-   f_Cdn_ocn      = 'x'
-/
-
-&icefields_bgc_nml
-/
-eof
+# Prepare local variables for use in ice_in_template from UFSWM
+# setup_nml section
+local SYEAR=${current_cycle:0:4}
+local SMONTH=${current_cycle:4:2}
+local SDAY=${current_cycle:6:2}
+local SECS=${sec}
+local DT_ICE=${ICETIM}
+local NPT=${npt}  # TBT [To be templated in ice_in_template; UFSWM sets to 999]
+local CICERUNTYPE=${runtype}
+local RUNID="unknown"
+local USE_RESTART_TIME=${use_restart_time}
+local RESTART_DIR="./CICE_RESTART/"  # TBT [UFSWM sets to './RESTART/']
+local RESTART_FILE="cice_model.res"  # TBT [UFSWM sets to 'iced']
+local DUMPFREQ=${dumpfreq}
+local DUMPFREQ_N=${dumpfreq_n}
+local DIAG_FREQ=6  # The template variable should be DIAGFREQ, not DIAG_FREQ (it goes against other variable conventions used)
+local HISTFREQ_N="0, 0, ${FHOUT}, 1, 1"  # TBT [UFSWM sets to '0, 0, 6, 1, 1']
+local CICE_HIST_AVG=${cice_hist_avg}
+local HISTORY_DIR="./CICE_OUTPUT/"  # TBT [UFSWM sets to './history/']
+local INCOND_DIR="./CICE_OUTPUT/"  # TBT [UFSWM sets to './history/']
+# grid_nml section
+local CICEGRID="${ice_grid_file}"
+local CICEMASK="${ice_kmt_file}"
+local GRIDATM="???"  # What are the global-workflow definitions for these? What are CICE defaults
+local GRIDOCN="???"  # What are the global-workflow definitions for these? What are CICE defaults
+local GRIDICE="???"  # What are the global-workflow definitions for these? What are CICE defaults
```

Comment:
@NeilBarton-NOAA 
Can you answer what the appropriate values for these variables are in the workflow?

Reply:
# default CICE B-grid, ATM and OCN are provided by cap on A grid
export GRIDATM=A
export GRIDOCN=A
export GRIDICE=B

Reply:
Thanks @NeilBarton-NOAA 
I will set these in the workflow.

Reply:
Included in commit [b7c7ed4](https://github.com/NOAA-EMC/global-workflow/pull/2051/commits/b7c7ed4551febd377b58006856b1d5e2e85c05db)